Streams/StreamWriter.cpp

```diff
@@ -0,0 +1,3 @@
+#include "StreamWriter.h"
+
+StreamWriter::~StreamWriter() { }
```

Comment:
StreamWriter is an interface, so we could reasonably omit a .cpp file. It's possible to allow for an inline empty destructor in the header file even though it's declared pure virtual. The pure virtual means that method doesn't require an implementation, and marks the class as abstract. It does not mean the function does not have an implementation. A rather subtle distinction.